### PR TITLE
Added mgird ogrid to jax.numpy 

### LINKED ISSUE #21720: mgrid

### LINKED ISSUE #21721: ogrid

### PR DESCRIPTION
**Pull request by ShreyanshBardia**
closes #21720
closes #21721

### CODE REVIEW

**Comment by ShreyanshBardia** [comment]
Hi @paulaehab, this PR is ready for review

**Comment by paulaehab** [comment]
Hi @ShreyanshBardia  thanks for contributing and sorry for late reply could you please add one function only to the PR not two functions?

**Comment by ShreyanshBardia** [comment]
Hi @paulaehab, the functions added depend on the same parent class and thus I thought it would be better to add both of them together, since the functions have minimal code of themselves, majority of it belonging to the Parent class. If I'll make two separate PR's I'll be adding same parent class again. But if you still think it's a better idea to add them separately, then I'll do so, please lemme know. Thanks!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
